src/ItemList.js

```diff
@@ -22,10 +45,25 @@ function ItemList(props) {
       )}
       {props.list && props.list.length > 0 && (
         <>
+          {/* [filter-list] 1. Add text field above shopping list */}
+          <label htmlFor="filter">
+            Filter Items
+            <input
+              id="filter"
+              type="text"
+              placeholder="Start typing here..."
+              value={query}
+              onChange={changeHandler}
+            />
+          </label>
+          {/* [filter-list] 4. Add 'X' button for user to clear query field */}
+          {query.length !== 0 && <button onClick={clickHandler}>X</button>}
```

Comment:
Should there be some label for screen readers on this button, like `aria-label="clear field"`?

Reply:
I'm never sure when to add aria labels, but this sounds like it could be a good idea. Will read up on this.